capstone/src/main/java/com/google/sps/servlets/StudentServlet.java

```diff
@@ -26,4 +30,19 @@ private static String convertToJsonUsingGson(Student student) {
     String json = gson.toJson(student);
     return json;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get student object based on the logged in email
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    Student student = PrototypeStudents.PROTOTYPE_STUDENTS.get(userEmail);
```

Comment:
So here, it looks like you're defining a local "student" variable. This means any changes you make to it won't be reflected permanently (which I'm assuming is the point of this change).

The point of hard-coded/explicitly defined Student objects in your prototype files is to have something to reference when the object is called. However, this means you won't be able to make changes to them that persist without some more-formal storage approach like using Datastore. This is a drawback of hardcoding.

So for the prototype, the actual behavior of add/remove isn't quite necessary (unless you have a working Datastore to access/modify objects). You can (and should still have the button), and can maybe still have other simple functionality like the redirection. But hard-coded objects are meant to be unmodified and the code shouldn't do this. Instead, maybe adding a TODO to implement this once Datastore is working

Reply:
Ah, I see. I was thinking of including the functionality (just to see what it would actually look like if the user would to click "Leave") and just changing it to access the Student object from Datastore once that was implemented, but this makes sense! I just took out the actual removing of the club from the Student's club list and instead added comments for all the doPost functionalities. Thanks for bringing this up!